Sources/CodeEditTextView/STTextViewController.swift

```diff
@@ -309,6 +312,67 @@ public class STTextViewController: NSViewController, STTextViewDelegate, ThemeAt
         }
     }
 
+    private func updateCursorPosition(with textLayoutManager: NSTextLayoutManager) {
+            /// Current cursor location as NSTextLocation
+            guard let insertionPointLocation = textLayoutManager.insertionPointLocation else { return }
+
+            var lineWrapsCount = 0
+
+            // TODO: Find a more performant approach, that does not scale with line count
+            /// Count the line wraps prior to the cursor line
+            textLayoutManager.enumerateTextLayoutFragments(from: textLayoutManager.documentRange.location,
+                                                           options: [.ensuresLayout, .ensuresExtraLineFragment]
+            ) { textLayoutFragment in
+
+                guard let cursorTextLineFragment = textLayoutManager.textLineFragment(at: insertionPointLocation)
+                else { return false }
+
+                /// Check whether the textLayoutFragment has line wraps
+                if textLayoutFragment.textLineFragments.count > 1 {
+                    for lineFragment in textLayoutFragment.textLineFragments {
+                        lineWrapsCount += 1
+                        /// Do not count lineFragments after the lineFragment where the cursor is placed
+                        if lineFragment == cursorTextLineFragment { break }
+                    }
+
+                    /// The first lineFragment will be counted as an actual line
+                        lineWrapsCount -= 1
+                }
+
+                if textLayoutFragment.textLineFragments.contains(cursorTextLineFragment) {
+                    return false
+                }
+                return true
+            }
+
+            /// Translate to line and column value
+            textLayoutManager.enumerateTextSegments(in: NSTextRange(location: insertionPointLocation),
+                                                    type: .standard,
+                                                    options:
+                                                        [.rangeNotRequired,
+                                                         .upstreamAffinity]
+            ) { _, textSegmentFrame, _, _ -> Bool
```

Comment:
Please try to use multi-line indents for arguments like this:

```suggestion
            textLayoutManager.enumerateTextSegments(
                in: NSTextRange(location: insertionPointLocation),
                type: .standard,
                options: [.rangeNotRequired, .upstreamAffinity]
            ) { _, textSegmentFrame, _, _ -> Bool
```

Reference: https://github.com/CodeEditApp/CodeEdit/wiki/Code-Style#parameter-alignment